examples/Example13/electrons.cu

```diff
@@ -0,0 +1,341 @@
+
+    // Get result into variables.
+    double geometricalStepLengthFromPhysics = theTrack->GetGStepLength();
+    // The phyiscal step length is the amount that the particle experiences
+    // which might be longer than the geometrical step length due to MSC. As
+    // long as we call PerformContinuous in the same kernel we don't need to
+    // care, but we need to make this available when splitting the operations.
+    // double physicalStepLength = elTrack.GetPStepLength();
+    int winnerProcessIndex = theTrack->GetWinnerProcessIndex();
+    // Leave the range and MFP inside the G4HepEmTrack. If we split kernels, we
+    // also need to carry them over!
+
+    // Check if there's a volume boundary in between.
+    double geometryStepLength;
+    vecgeom::NavStateIndex nextState;
+    if (BzFieldValue != 0) {
+      geometryStepLength = fieldPropagatorBz.ComputeStepAndPropagatedState</*Relocate=*/false, BVHNavigator>(
+          currentTrack.energy, Mass, Charge, geometricalStepLengthFromPhysics, currentTrack.pos, currentTrack.dir,
+          currentTrack.navState, nextState);
+    } else {
+      geometryStepLength =
+          BVHNavigator::ComputeStepAndNextVolume(currentTrack.pos, currentTrack.dir, geometricalStepLengthFromPhysics,
+                                                 currentTrack.navState, nextState, kPush);
+      currentTrack.pos += geometryStepLength * currentTrack.dir;
+    }
+
+    // Set boundary state in navState so the next step and secondaries get the
+    // correct information (currentTrack.navState = nextState only if relocated
+    // in case of a boundary; see below)
+    currentTrack.navState.SetBoundaryState(nextState.IsOnBoundary());
+
+    // Propagate information from geometrical step to MSC.
+    theTrack->SetDirection(currentTrack.dir.x(), currentTrack.dir.y(), currentTrack.dir.z());
+    theTrack->SetGStepLength(geometryStepLength);
+    theTrack->SetOnBoundary(nextState.IsOnBoundary());
+
+    // Apply continuous effects.
+    bool stopped = G4HepEmElectronManager::PerformContinuous(&g4HepEmData, &g4HepEmPars, &elTrack, &rnge);
+
+    // Collect the direction change and displacement by MSC.
+    const double *direction = theTrack->GetDirection();
+    currentTrack.dir.Set(direction[0], direction[1], direction[2]);
+    if (!nextState.IsOnBoundary()) {
+      const double *mscDisplacement = mscData->GetDisplacement();
+      vecgeom::Vector3D<Precision> displacement(mscDisplacement[0], mscDisplacement[1], mscDisplacement[2]);
+      const double dLength2            = displacement.Length2();
+      constexpr double kGeomMinLength  = 5 * copcore::units::nm;          // 0.05 [nm]
+      constexpr double kGeomMinLength2 = kGeomMinLength * kGeomMinLength; // (0.05 [nm])^2
+      if (dLength2 > kGeomMinLength2) {
+        const double dispR = std::sqrt(dLength2);
+        // Estimate safety by subtracting the geometrical step length.
+        safety -= geometryStepLength;
```

Comment:
I think we should subtract from the cached safety value here

Reply:
`safety` is the value computed at the beginning of the step, I don't understand where I should subtract from otherwise. (This case is only to avoid calling `BVHNavigator::ComputeSafety` in cases where the safety is much longer than the step length and we can apply the displacement right away.)